### PR TITLE
Support Detach.

### PR DESCRIPTION
**Pull request by rajkumar42**
Sending "stopped" message with reason as "disconnected" to avoid the confusion.

### CODE REVIEW

**Comment by rajkumar42** [comment]
@edumunoz

**Comment by edumunoz** [comment]
Wouldn't `HandleThreadGroupExited` still be called, causing the `"*stopped,reason=""exited"""` string to be processed too?

File: src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs [hunk 1]
```diff
@@ -1167,7 +1167,12 @@ public void Terminate()
             ScheduleStdOutProcessing(@"*stopped,reason=""exited"",exit-code=""42""");
         }
 
-        public void Detach() { }
+        public void Detach()
+        {
+            // Fake stopped message to send ProgramDestroyEvent.
+            ScheduleStdOutProcessing(@"*stopped,reason=""disconnected""");
```
Comment (edumunoz) [inline src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs]: I don't see the reason `"disconnected"` being handled on the `OnStopped` handler (in Debugger.cs).

Shouldn't it cause the debugger to exit (`-gdb-exit`)?

Comment (rajkumar42) [inline src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs]: Sorry, I missed the file during comment. Have added it now.

**Comment by gregg-miskelly** [comment]
:+1:

**Pull request merged**